kernel/src/hil/flash.rs

```diff
@@ -142,3 +145,154 @@ pub trait Client<F: Flash> {
     /// Flash erase complete.
     fn erase_complete(&self, error: Error);
 }
+
+// *** Interfaces for hardware with flash memory protection support ***
+
+/// Define the basic region permissions for flash memory protection.
+#[derive(PartialEq, Debug)]
+pub struct FlashMPBasicConfig {
+    /// Region can be read.
+    pub read_en: bool,
+    /// Region can be programmed.
+    pub write_en: bool,
+}
+
+/// Memory access control protection for flash. For hardware that supports
+/// flash memory protection, this can be used to implement the relevant functionality.
+///
+/// Requires the Flash interface to have been implemented as well.
```

Comment:
In the context of the possibility for additional future traits as alternatives to the current `Flash` trait, do we need to require this?

Reply:
Wouldn't we want to make sure flash is implemented before flash memory protection is though?

Reply:
I don't think it could really hurt for something to implement this trait without implementing flash, it just wouldn't be useful. And if https://github.com/tock/tock/pull/2993 is merged, it is very possible there could be something that implements `BlockStorage` but not `Flash` which still might benefit from flash memory protection